projects/packages/forms/src/blocks/step/edit.js

```diff
@@ -0,0 +1,177 @@
+import { useBlockProps, useInnerBlocksProps, InnerBlocks } from '@wordpress/block-editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as singleStepStore } from '../../store/preview-store';
+import AddStepControls from '../contact-form/components/add-step-controls';
+import StepControls from '../contact-form/components/step-controls';
+import useFormSteps from '../shared/hooks/use-form-steps';
+import useParentFormClientId from '../shared/hooks/use-parent-form-client-id';
+import AttributesControls from './attributes-controls';
+
+import './editor.scss';
+
+// Define allowed blocks directly in this file to break circular dependency
+const ALLOWED_BLOCKS = [
+	'jetpack/field-text',
+	'jetpack/field-name',
+	'jetpack/field-email',
+	'jetpack/field-url',
+	'jetpack/field-date',
+	'jetpack/field-telephone',
+	'jetpack/field-number',
+	'jetpack/field-textarea',
+	'jetpack/field-checkbox',
+	'jetpack/field-checkbox-multiple',
+	'jetpack/field-option-checkbox',
+	'jetpack/field-radio',
+	'jetpack/field-option-radio',
+	'jetpack/field-select',
+	'jetpack/field-consent',
+	'jetpack/form-step-navigation',
+	'jetpack/step-divider',
+	'core/audio',
+	'core/columns',
+	'core/group',
+	'core/heading',
+	'core/html',
+	'core/image',
+	'core/list',
+	'core/paragraph',
+	'core/row',
+	'core/separator',
+	'core/spacer',
+	'core/stack',
+	'core/subhead',
+	'core/video',
+];
+
+// Replace the constant with a function
+const getStepTemplate = ( previousStepBlocks = [] ) => {
+	// Check if previous step has navigation
+	const hasNavigation = previousStepBlocks.some(
+		block => block.name === 'jetpack/form-step-navigation'
+	);
+
+	if ( hasNavigation ) {
+		return [
+			[ 'core/paragraph', {} ],
+			[ 'jetpack/form-step-navigation', {} ],
+		];
+	}
+};
+
+function StepBreak( { stepLabel, currentIndex } ) {
+	// Translators: %d is the step number (1, 2, 3, etc.)
+	let stepName = sprintf( __( 'Step %d', 'jetpack-forms' ), currentIndex + 1 );
+
+	if ( stepLabel !== '' || stepLabel !== 'Step' ) {
+		// Translators: %1$d is the step number (1, 2, 3, etc.), %2$s is the step label
+		stepName = sprintf( __( 'Step %1$d - %2$s', 'jetpack-forms' ), currentIndex + 1, stepLabel );
```

Comment:
Let's use endash for number ranges.

```suggestion
		stepName = sprintf( __( 'Step %1$d–%2$s', 'jetpack-forms' ), currentIndex + 1, stepLabel );
```